### ISSUE TITLE
VOTING: When user has not joined a public topic, can't vote

### ISSUE THREAD

**Issue by BeccaMelhuish** [citizenos/citizenos-fe#1714]
**What is the problem?**
A user contacted @ssin1901 (Support) to say that they were unable to vote on a public topic. The solution was that they had to join the topic in order to be able to vote. 

Checking this myself, I see that the UX is really unclear if you enter the public topic without having joined it. You cannot click the vote options to select them, and the 'submit button' is greyed out, but there are no hints to the user as to why this may be: 
![image](https://github.com/user-attachments/assets/99ef0ea5-d2a2-4e7c-aea1-fc933f02526b)

**Why is this a problem?**
Causes confusion and barriers to voting.

**Possible solution.**

**- User is trying to vote on a public topic they haven't yet joined, and is already logged in.** 
Seeing as we decided we would automatically add people to a public topic once they've engaged in any way, without them needing to actively join it, then for logged there should be no difference in the UX for those who have joined and those who haven't (unless there is some technical need for there to be). 

**- User is trying to vote on a public topic they haven't yet joined, and is not yet logged in.**
They should be able to select their vote option(s) and click the 'Submit vote' button as if they were logged in, but upon clicking the button, it should bring up the 'Please log in to continue' box. After they have logged in or registered they should be brought back to where they were, and be notified that their vote has been submitted.

Tagging @kevincrepin @Lidziya-p @TiinaDUX for info or comment if needed but I think the UX is fairly clear :)

**Comment by ilmartyrk** [comment]
Will have to check, but it seems to be an easy fix so estimating 30mins

**Comment by kevincrepin** [comment]
Yes solution is clear and I'm not sure but I thought we had previously already agreed that – like you say @BeccaMelhuish – the user would automatically join a topic when they participate in it. So all actions should be available, not greyed out.

**Comment by BeccaMelhuish** [comment]
Closing as is being fixed via #307
